src/main/java/fi/aalto/cs/apluscourses/presentation/exercise/ExerciseViewModel.java

```diff
@@ -31,6 +31,15 @@ public boolean isSubmittable() {
         && !"Assignment  debugger".equals(name) && (maxSubmissions == 10 || maxSubmissions == 0);
   }
 
+  /**
+   * Returns {@code true} if the user has gotten maximum points, {@code false} otherwise.
+   */
+  public boolean isCompleted() {
+    int maxPoints = getModel().getMaxPoints();
+    int userPoints = getModel().getUserPoints();
+    return userPoints == maxPoints;
+  }
```

Comment:
Does this mark all optional practice assignments as completed, since they have 0 max points? Perhaps an additional condition that the exercise has at least one submission

Reply:
Furthermore, this logic would suit better in the model class (`Exercise`) because it's not about presentation but a property of the domain object (an exercise/assignment) itself. (Yeah, I know, we haven't been completely consistent in these matters ourselves. :smile:)

Reply:
And because @superseacat is on vacation, I say this on his behalf: Could we have unit tests? :laughing: